apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -245,37 +250,52 @@ export default function LearningGoals({
 
   useEffect(() => {
     if (studentLevelInfo && learningGoals) {
-      learningGoals.forEach((learningGoal, index) => {
-        const body = JSON.stringify({
-          userId: studentLevelInfo.user_id,
-          learningGoalId: learningGoal.id,
-        });
-        HttpClient.post(
-          `${base_teacher_evaluation_endpoint}/get_or_create_evaluation`,
-          body,
-          true,
-          {
-            'Content-Type': 'application/json',
-          }
-        )
-          .then(response => response.json())
-          .then(json => {
-            learningGoalEvalIds.current[index] = json.id;
-            if (json.feedback) {
-              teacherFeedbacks.current[index] = json.feedback;
-            }
-            if (json.understanding >= 0 && json.understanding !== null) {
-              understandingLevels.current[index] = json.understanding;
-            }
-            if (index === currentLearningGoal) {
-              setDisplayFeedback(teacherFeedbacks.current[currentLearningGoal]);
-              setDisplayUnderstanding(
-                understandingLevels.current[currentLearningGoal]
-              );
+      // Set our current idea of the feedback immediately
+      setDisplayFeedback(teacherFeedbacks.current[currentLearningGoal]);
+      setDisplayUnderstanding(understandingLevels.current[currentLearningGoal]);
+
+      // Only load prior learning goal feedback once
+      learningGoals
+        .filter((learningGoal, index) => {
+          return !teacherFeedbacksLoaded.current[index];
+        })
+        .forEach((learningGoal, index) => {
+          const body = JSON.stringify({
+            userId: studentLevelInfo.user_id,
+            learningGoalId: learningGoal.id,
+          });
+          HttpClient.post(
+            `${base_teacher_evaluation_endpoint}/get_or_create_evaluation`,
+            body,
+            true,
+            {
+              'Content-Type': 'application/json',
             }
-          })
-          .catch(error => console.error(error));
-      });
+          )
+            .then(response => response.json())
+            .then(json => {
+              learningGoalEvalIds.current[index] = json.id;
+              if (json.feedback) {
+                teacherFeedbacks.current[index] = json.feedback;
+              }
+              teacherFeedbacksLoaded.current[index] = true;
+              if (json.understanding >= 0 && json.understanding !== null) {
+                understandingLevels.current[index] = json.understanding;
+              }
+              if (index === currentLearningGoalRef.current) {
+                setDisplayFeedback(
+                  teacherFeedbacks.current[currentLearningGoalRef.current]
+                );
+                setLoaded(
+                  teacherFeedbacksLoaded.current[currentLearningGoalRef.current]
+                );
+                setDisplayUnderstanding(
+                  understandingLevels.current[currentLearningGoalRef.current]
```

Comment:
I think this block is worth highlighting with a brief comment explaining its role in preventing race conditions

Reply:
Agreed. Though. the 'brief' you threw in there is the real challenge.

Reply:
"make sure the currently visible learning goal is updated when the data fetch completes"